activesupport/lib/active_support/hash_with_indifferent_access.rb

```diff
@@ -159,6 +157,20 @@ def key?(key)
     alias_method :has_key?, :key?
     alias_method :member?, :key?
 
+
+    # Same as <tt>Hash#[]</tt> where the key passed as argument can be
+    # either a string or a symbol:
+    #
+    #   counters = ActiveSupport::HashWithIndifferentAccess.new
+    #   counters[:foo] = 1
+    #
+    #   counters['foo'] # => 1
+    #   counters[:foo]  # => 1
+    #   counters[:zoo]  # => nil
+    def [](key)
```

Comment:
@sgrif there's a performance hit on reads from existing keys. The current implementation relies on overwriting `default` which is called only if a key doesn't exist but that leads to different behaviour between HWIA and Hash.